documentation/FrontEndTutorial.md

```diff
@@ -0,0 +1,747 @@
+<!--===- documentation/FrontEndTutorial.md 
+  
+   Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+   See https://llvm.org/LICENSE.txt for license information.
+   SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+  
+-->
+# Introduction
+I recently added a semantic check to the f18 compiler front end.  This document
+describes my thought process and the resulting implementation.
+
+For more information about the compiler, start with the 
+[compiler overview](Overview.md).
+
+# Problem definition
+
+In the 2018 Fortran standard, section 11.1.7.4.3, paragraph 2, states that:
+
+```
+Except for the incrementation of the DO variable that occurs in step (3), the DO variable 
+shall neither be redefined nor become undefined while the DO construct is active.
+```
+One of the ways that DO variables might be redefined is if they are passed to
+functions with dummy arguments whose ```INTENT``` is ```INTENT(IN)``` or
```

Comment:
INTENT(OUT)?

Reply:
Good catch.

Reply:
You can use single backquotes (grave accents) for inline code.

Reply:
Thanks!